CHANGELOG.md

```diff
@@ -41,6 +41,49 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## Unreleased
 
+* [#1435] `x/tokenfactory` create denom fee for spam resistance 
+* [#1429] solver for multi-asset CFMM
+* [#1253] Add lockup duration edit method
+* [#1253] Add lockup duration edit method 
+* [#1312] Stableswap: Createpool logic 
+* [#1230] Stableswap CFMM equations  
+## [v8.0.0 - Emergency proposals upgrade](https://github.com/osmosis-labs/osmosis/releases/tag/v8.0.0)
+
+This upgrade is a patch that must be hard forked in, as on-chain governance of Osmosis approved proposal [227](https://www.mintscan.io/osmosis/proposals/227) and proposal [228](https://www.mintscan.io/osmosis/proposals/228).
+
+This upgrade includes:
+
+* Adding height-gated AnteHandler message filter to filter unpooling tx pre-upgrade.
+* At block height 4402000 accelerates prop 225, which in turn moves incentives from certain pools according to props 222-224
+* Adds a msg allowing unpooling of UST pools. 
+  * This procedure is initiated by whitelisting pools 560, 562, 567, 578, 592, 610, 612, 615, 642, 679, 580, 635. 
+  * Unpooling allows exiting whitelisted pools directly, finish unbonding duration with the exited tokens instead of having to wait unbonding duration to swap LP shares back to collaterals. 
+  * This procedure also includes locks that were already unbonding pre-upgrade and locks that were superfluid delegated.
+
+Every node should upgrade their software version to v8.0.0 before the upgrade block height 4402000. If you use cosmovisor, simply swap out the binary at upgrades/v7/bin to be v8.0.0, and restart the node. Do check cosmovisor version returns v8.0.0
+
+### Features 
+
+* {Across many PRs} Initiate emergency upgrade 
+* [#1481] Emergency upgrade as of prop [226] (https://www.mintscan.io/osmosis/proposals/226) 
+* [#1482] Checking Whitelisted Pools contain UST 
+* [#1486] Update whitelisted pool IDs
+* [#1262] Add a forceprune command to the binaries, that prunes golevelDB data better
+* [#1154] Database stability improvements
+* [#840] Move lock.go functions into iterator.go, lock_refs.go and store.go
+* [#916] And a fn for Unbond and Burn tokens
+* [#908] Superfluid slashing code
+* [#904] LockAndSuperfluidDelegate
+### Minor improvements & Bug Fixes
+
+* [#1428] fix: pool params query (backport #1315)
+* [#1390] upgrade sdk to v0.45.0x-osmo-v7.9
+* [#1087] Test improvisation for Superfluid (backport #1070)
+* [#1022] upgrade iavl to v0.17.3-osmo-v4
+* [#958] Fix queries and add codec for msgsuperfluidunbondlock
+* [#958] Update the risk factor from 5% to 50%
+* [#894] x/superfluid: Delegate bug fixes and refactors
```

Comment:
```suggestion
```

Reply:
Sorry about that @ValarDragon. I got a little confused. Is there a way to check which PRs and bug fixes vs improvements without going thro them individually?